hack/install-controller-gen.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# Copyright The Shipwright Contributors
+# 
+# SPDX-License-Identifier: Apache-2.0
+
+#
+# Installs controller-gen utility via "go get".
+#
+
+set -eu
+
+# kind version
+CONTROLLER_GEN_VERSION="${CONTROLLER_GEN_VERSION:-v0.4.1}"
+
+if [ ! -f "${GOPATH}/bin/controller-gen" ] ; then
+    echo "# Installing controller-gen..."
+    go get sigs.k8s.io/controller-tools/cmd/controller-gen@"${CONTROLLER_GEN_VERSION}"
+fi
```

Comment:
Until we can use the Go 1.16 improved go get/install that does not fiddle with the current Go modules file, I would suggest to do the installation part in a temporary directory.

```suggestion
if [ ! -f "${GOPATH}/bin/controller-gen" ] ; then
    echo "# Installing controller-gen..."
    pushd "$(mktemp -d)" >/dev/null 2>&1
    go get sigs.k8s.io/controller-tools/cmd/controller-gen@"${CONTROLLER_GEN_VERSION}"
    popd >/dev/null 2>&1
fi
```

Reply:
Thanks, Matthias. I updated the code according to your suggestion, BTW, do you think if it's needed to clean up this temporary dir after installation? Thanks.